PWGCF/DataModel/FemtoDerived.h

```diff
@@ -185,25 +185,36 @@ namespace femtodreamparticleMC
 {
 /// Distinuishes the different particle origins
 enum ParticleOriginMCTruth {
-  kPrimary,   //! Primary track or V0
-  kDaughter,  //! Particle from a decay
-  kMaterial,  //! Particle from a material
-  kNotPrimary //! Particle from a material
+  kPrimary,    //! Primary track or V0
+  kDaughter,   //! Particle from a decay
+  kMaterial,   //! Particle from a material
+  kNotPrimary, //! Particle from a material
+  kFake,
+  kDaughterLambda,
+  kDaughterSigmaplus
 };
 
-DECLARE_SOA_INDEX_COLUMN(FemtoDreamParticle, femtoDreamParticle);
+// DECLARE_SOA_INDEX_COLUMN(FemtoDreamCollision, femtoDreamCollision);
+// DECLARE_SOA_INDEX_COLUMN(FemtoDreamParticle, femtoDreamParticle);
 DECLARE_SOA_COLUMN(PartOriginMCTruth, partOriginMCTruth, uint8_t); //! Origin of the particle, according to femtodreamparticle::ParticleOriginMCTruth
 DECLARE_SOA_COLUMN(PDGMCTruth, pdgMCTruth, int);                   //! Particle PDG
+DECLARE_SOA_COLUMN(PtTruth, ptTruth, float);                       //! p_T of the generated particle (GeV/c)
+DECLARE_SOA_COLUMN(EtaTruth, etaTruth, float);                     //! Eta of the generated particle
+DECLARE_SOA_COLUMN(PhiTruth, phiTruth, float);                     //! Phi of the generated particle
```

Comment:
The intention of the original author was to add columns containing MC truth to the table holding information about the reconstructed particle (I am refering to the tables of the femtodream particle table). Since in femtoscopy we only care about purity, we do not need the information about generated particles that are not being reconstructed (or reconstructed particles that do not pass minimal cuts). As far as I understand, each column needs a unique name, so that is why we have pt-pttruth, eta-etatruth.

If I understand your suggestions correctly, we should implement a standalone table holding the MC truth and pass both tables through the framework instead of joining the MC information to the existing table?

Reply:
No. What I'm suggesting is to get rid of these new column you are defining in here
The for the MC table declaration in L.209 use the columns declared in L.73-75, i.e.
```
DECLARE_SOA_COLUMN(Pt, pt, float);                    //! p_T (GeV/c)
DECLARE_SOA_COLUMN(Eta, eta, float);                  //! Eta
DECLARE_SOA_COLUMN(Phi, phi, float);                  //! Phi
```
Then, as far a you use only this columns in the routines, you don't need to replicate any code, the template mechanism ensure that things are done correctly. 
Of course you keep populating the MC table in the same way you do it now. Nothing needs to be changed.
The only things
- remove the additional columns
- use the same columns as for reco and data
- remove the replicated routines
- call the same templated routines as for reco and data

As said, this will work for the cases shown in this PR
It should work also in the general case but when something specific is need which differentiates the MC table from the Reco/Data table

Reply:
Sorry, Victor, I think we are talking passed each other or maybe the lableing of our columns is missleading (or there is something else I am missing, because implementing your changes would, I think, require quite some changes to the framework)

The columns ptTruth, etaTruth and phiTruth, which we use to compose the FemtoDreamParticlesMC table, are not meant to be used standalone, but must be joined with the FemtoDreamParticles table when we do MC processing. That means one row in the joined table (i.e. one reconstructed particle) will have columns containing values for pt, eta, phi (on the reconstructed level) and ptTruth, etaTruth and phiTruth (on the generator level), which will in general be different values. Like I said before, in femtoscopy we only care about the MC truth of particles that are reconstructed and who pass the minimal selections. In our producer task we only ever iterate over reconstructed particles, extract the information we need (including MC truth, if available) and write them into our FemtoDreamParticleTable. In principle we could add just 3 more columns to FemtoDreamParticleTable for the MC truth and just set them to some default value when there is no MC truth, but we thought it would be smarter if write these values into seperate columns (or table) which we can join later on demand. (Strictly speaking, we would need 4 columns, since we also need to the know the particle's origin).

That was also the reason why we orignially implemented two versions for the getkstar method, since we used different columns (of the same particle) for the computation of kstar and get different values (a lazy solution, I admit).

I think a straight forward solution to the code duplication problem would be go get rid off the template of the getkstar and similar methods and don't pass the particles and just pass the kinematic information of both particles (i.e. pt,eat,phi,mass per particle) directly as floats. I already implemented a patch for that, if you are fine with that solution.

Reply:
Well, in my opinion both are bad design decisions
- joining the reco and gen tables forces you to replicate code
- getting rid of the templating forces you to store table values in other places in memory which penalize the execution and is highly not recommended 

The right way of doing it should be the one use in the overall data model, i.e. having a table which addresses the MC particles and which is joinable to the reconstructed tracks table. For the kinematic variables the MC and the tracks tables share the column names so that the functions that manipulate them are easily templated. Of course, in the equivalent MC table in your case you only put what you want, i.e. only MC particles associated to reconstructed tracks. The advantage is that if you run on MC productions on